### PR TITLE
Update storage to work with all three entities

### LINKED ISSUE #90: Update storage to work with all three entities

### PR DESCRIPTION
**Pull request by lkldev**
Resolves #90

This PR will stop the program from reading the database created by AB3. We can soon remove traces of AB3 after we have replicated the test cases for the various classes.

This PR includes:
* Update jackson package to support Optional in Json Adapted Entities
* Create Json classes to serialize storage into TAssist object
* Update Storage and UserPrefs to use tassist.json as storage
* Update common classes to use the storage created

### CODE REVIEW

**Review by Gernene — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/storage/TAssistStorage.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public interface TAssistStorage {
+
+    /**
+     * Returns the file path of the data file.
+     */
+    Path getTAssistFilePath();
+
+    /**
+     * Returns TAssist data as a {@link ReadOnlyTAssist}.
+     *   Returns {@code Optional.empty()} if storage file is not found.
```
Comment (jxt00) [inline src/main/java/seedu/address/storage/TAssistStorage.java:22]: Should remove the extra space

Comment (lkldev) [reply src/main/java/seedu/address/storage/TAssistStorage.java]: Resolved in latest commit

**Review by jxt00 — APPROVED** [review]
LGTM 👍

**Pull request merged**